test/bluebird.test.js

```diff
@@ -94,10 +100,189 @@ describe('bluebird', function () {
         return Promise.reject();
       }, {
         max: 3,
+        backoffBase: 0,
         timeout: 1500
       })).to.eventually.be.rejectedWith(Promise.TimeoutError).then(function () {
         expect(count).to.equal(3);
       });
     });
   });
-});
\ No newline at end of file
+  
+  describe('match', function () {
+    it('should continue retry while error is equal to match string', function () {
+        return expect(retry(function () {
+        count++;
+
+        if (count < 4) {
+          return Promise.reject(soRejected);
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15,
+        backoffBase: 1,
+        match: soRejected
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(4);
+      });
+    });
```

Comment:
I realize my original tests use count aswell, but i've gotten wiser since then.
We should add the packages `sinon`, `sinon-chai` and `sinon-as-promised'.

This test could then be refactored to:

```js
var callback = this.sinon.stub();
callback.resolves(soResolved);
callback.onCall(4).rejects(soRejected);
return expect(retry(callback)).to.eventually.equal(soResolved).then(function () {
  expect(callback.callCount).to.equal(4);
});
```